testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/SinglenodeCompatibility.java

```diff
@@ -0,0 +1,129 @@
+        extends EnvironmentProvider
+{
+    private static final int SERVER_PORT = 18080;
+    private static final String COMPATIBILTY_TEST_CONTAINER_NAME = "compatibility-test-coordinator";
+
+    private final DockerFiles dockerFiles;
+    private final DockerFiles.ResourceProvider configDir;
+    private final PortBinder portBinder;
+
+    private Config extraConfig;
+
+    @Inject
+    public SinglenodeCompatibility(Standard standard, Hadoop hadoop, DockerFiles dockerFiles, PortBinder portBinder)
+    {
+        super(ImmutableList.of(standard, hadoop));
+        this.dockerFiles = dockerFiles;
+        this.configDir = dockerFiles.getDockerFilesHostDirectory("conf/environment/singlenode-compatibility");
+        this.portBinder = portBinder;
+    }
+
+    @Override
+    public void extendEnvironment(Environment.Builder builder)
+    {
+        configureCompatibilityTestContainer(builder, extraConfig);
+    }
+
+    private void configureCompatibilityTestContainer(Environment.Builder builder, Config config)
+    {
+        String configDir = getConfigurationDirectory(config.getCompatibilityTestDockerImage());
+        DockerContainer container = new DockerContainer(config.getCompatibilityTestDockerImage(), COMPATIBILTY_TEST_CONTAINER_NAME)
+                .withNetworkAliases(COMPATIBILTY_TEST_CONTAINER_NAME + ".docker.cluster")
+                .withExposedLogPaths("/var/trino/var/log", "/var/log/container-health.log")
+                .withCopyFileToContainer(forHostPath(dockerFiles.getDockerFilesHostPath("conf/presto/etc/jvm.config")), configDir + "jvm.config")
+                .withCopyFileToContainer(forHostPath(this.configDir.getPath("config.properties")), configDir + "config.properties")
+                .withCopyFileToContainer(forHostPath(this.configDir.getPath("hive.properties")), configDir + "catalog/hive.properties")
+                .withCopyFileToContainer(forHostPath(dockerFiles.getDockerFilesHostPath()), "/docker/presto-product-tests")
+                .withStartupCheckStrategy(new IsRunningStartupCheckStrategy())
+                .waitingForAll(forLogMessage(".*======== SERVER STARTED ========.*", 1), forHealthcheck())
+                .withStartupTimeout(Duration.ofMinutes(5));
+        builder.addContainer(container);
+        portBinder.exposePort(container, SERVER_PORT);
+    }
+
+    private String getConfigurationDirectory(String dockerImageName)
+    {
+        String version = DockerImageName.parse(dockerImageName).getVersionPart();
+        if ("351".equals(version)) {
+            // 351 has the Trino configuration at a different location
+            return "/usr/lib/trino/default/etc/";
+        }
+
+        String repository = DockerImageName.parse(dockerImageName).getRepository();
+        switch (repository) {
+            case "trinodb/trino":
+                return "/etc/trino/";
+            case "trinodb/presto":
+                return "/usr/lib/presto/default/etc/";
```

Comment:
in that image, the `/etc/presto` was supported, right?

Reply:
The docker entrypoint doesn't copy the other default config files in case an `/etc/presto` already exists. I can use `/etc/presto` and `/etc/trino` to simplify but then I'd have to provide all config files (jvm.config, node.properties etc.).